lib/authorizeAccessToken.js

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright (c) 2021 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as authorization from 'auth-header';
+import bedrock from 'bedrock';
+import * as JWT from '@digitalbazaar/minimal-jwt';
+import {tokenizers} from 'bedrock-tokenizer';
+import base64url from 'base64url-universal';
+
+const {util: {BedrockError}} = bedrock;
+
+/**
+ * Authorizes an incoming request for a route (parses Authorization: header,
+ * decodes access token, verifies it, checks the decoded token claims against
+ * the provided scope, sets the `req.clientMetadata` property).
+ *
+ * @param {object} options - Options hashmap.
+ * @param {string} options.scope - Scope required for the route guarded by this
+ *   `authorize()` usage.
+ * @param {Function} [options.customValidate] - Optional custom authorization
+ *   callback with the signature `async ({req, claims}) => {}`.
+ * @param {object} [options.cache] - Optional LRU cache to store the decoded
+ *   token claims). The cache API must be compatible with the lru-cache NPM
+ *   module.
+ * @param {string[]} options.validIssuers - List of allowed issuers of
+ *   access tokens (e.g. 'https://admin.sandbox.nacsav.com').
+ * @param {Function} options.loadClientRegistration -
+ * @returns {Function} Returns an Express.js middleware route handler.
+ */
+export function authorizeAccessToken({
+  scope, cache, validIssuers, loadClientRegistration, customValidate}) {
+  return async (req, res, next) => {
+    try {
+      // Extract access token from Authorization: header (throw otherwise)
+      let clientAccessToken;
+      try {
+        // parse throws if the authorization header is missing
+        const auth = authorization.parse(req.get('authorization'));
+        if(!(auth.scheme === 'Bearer' && auth.token)) {
+          throw new Error('NotAllowedError');
+        }
+        ({token: clientAccessToken} = auth);
+      } catch(e) {
+        throw new Error('Missing or invalid "authorization" header.');
+      }
```

Comment:
If these are not `BedrockError` they are going to be surfaced to the user via HTTP as generic 500 server failures.

Reply:
We're now using @interop/oauth2-errors error handler instead of BedrockError. So marking this resolved.